src/main/java/org/teleight/teleightbots/scheduler/SchedulerImpl.java

```diff
@@ -9,14 +9,11 @@
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 
-public class SchedulerImpl implements Scheduler {
+public final class SchedulerImpl implements Scheduler {
```

Comment:
I think the scheduler is one of those components that should be extensible by developers